protoc-gen-doc panics on any input.

Running `protoc-gen-doc` and giving it no input (pressing Ctrl+D in bash) results in a nil pointer dereference panic.

```
$ protoc-gen-doc
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0xc0b13]

goroutine 1 [running]:
github.com/golang/protobuf/protoc-gen-go/plugin.(*CodeGeneratorRequest).Reset(0x0)
	/src/github.com/golang/protobuf/protoc-gen-go/plugin/plugin.pb.go:48 +0x13
github.com/golang/protobuf/proto.Unmarshal(0xc2080d2400, 0x0, 0x200, 0x543eb8, 0x0, 0x0, 0x0)
	/src/github.com/golang/protobuf/proto/decode.go:299 +0x40
main.main()
	/src/sourcegraph.com/sourcegraph/prototools/cmd/protoc-gen-doc/main.go:83 +0x276

goroutine 5 [chan receive]:
github.com/golang/glog.(*loggingT).flushDaemon(0x488ca0)
	/src/github.com/golang/glog/glog.go:879 +0x78
created by github.com/golang/glog.init·1
	/src/github.com/golang/glog/glog.go:410 +0x2a7

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/src/runtime/asm_amd64.s:2232 +0x1
```

Looking into the code, I think it will panic for any input here:

```Go
var request *plugin.CodeGeneratorRequest
if err := proto.Unmarshal(data, request); err != nil {
	log.Fatal(err, ": failed to parse input proto")
}
```

Because `request` given to `proto.Unmarshal` is `nil`, and that will cause a nil pointer dereference.

I think it's a regression from #9, previously `proto.Unmarshal` was given a non-`nil` `Request`.

--- a/cmd/protoc-gen-doc/main.go
+++ b/cmd/protoc-gen-doc/main.go
@@ -79,7 +79,7 @@ func main() {
 	}
 
 	// Unmarshal the protoc generation request.
-	var request *plugin.CodeGeneratorRequest
+	request := &plugin.CodeGeneratorRequest{}
 	if err := proto.Unmarshal(data, request); err != nil {
 		log.Fatal(err, ": failed to parse input proto")
 	}

--- a/tmpl/generator.go
+++ b/tmpl/generator.go
@@ -141,7 +141,6 @@ func (g *Generator) SetRequest(r *plugin.CodeGeneratorRequest) error {
 // New returns a new generator for the given template.
 func New() *Generator {
 	return &Generator{
-		request:  &plugin.CodeGeneratorRequest{},
 		response: &plugin.CodeGeneratorResponse{},
 		registry: gateway.NewRegistry(),
 	}

